consensus/enclave/impl/src/lib.rs

```diff
@@ -462,44 +460,47 @@ impl ConsensusEnclave for SgxConsensusEnclave {
         };
 
         let total_fee: u64 = transactions.iter().map(|tx| tx.prefix.fee).sum();
-        let fee_minting_transaction = mint_aggregate_fee(&fee_tx_private_key, total_fee, blinding)?;
-
-        // The fee is created in the zero-th redacted transaction.
-        redacted_transactions.push(fee_minting_transaction);
+        let fee_output = mint_aggregate_fee(&fee_tx_private_key, total_fee, blinding)?;
 
-        // Redact each input transaction.
-        for tx in transactions {
-            redacted_transactions.push(tx.redact());
+        let mut outputs: Vec<TxOut> = Vec::new();
+        let mut key_images: Vec<KeyImage> = Vec::new();
+        for tx in &transactions {
+            outputs.extend(tx.prefix.outputs.iter().cloned());
+            key_images.extend(tx.key_images().iter().cloned());
         }
+        outputs.push(fee_output);
+
+        // TODO: sort outputs and key images.
```

Comment:
It might be best to do this sorting in a separate PR since reordering the outputs and key images will probably break a lot of unit tests.

Reply:
https://mobilecoin.atlassian.net/browse/MC-1337